libs/MeshKernelApi/src/MeshKernel.cpp

```diff
@@ -1893,6 +1893,86 @@ namespace meshkernelapi
         return lastExitCode;
     }
 
+    MKERNEL_API int mkernel_mesh2d_get_face_polygons(int meshKernelId, int numNodes, GeometryList& facePolygons)
```

Comment:
Syntactically,. C does not have references. Since this is a C API we should use a pointer. One added benefit is the ability to check if facePolygons is null and subsequently throw an exception. 
I don't think the unit tests cover such scenario.  Nothing to do for now, but something to think about.